drivers/input/input_cf1133.c

```diff
@@ -0,0 +1,367 @@
+		data->sensing_mode =
+			(buffer & (ONE_D_SENSING_CONTROL_BMSK << ONE_D_SENSING_CONTROL_SHFT)) >>
+			ONE_D_SENSING_CONTROL_SHFT;
+		LOG_DBG("sensing mode = %d ", data->sensing_mode);
+	} else {
+		data->touch_protocol_type = SITRONIX_A_TYPE;
+		LOG_DBG("i2c protocol = %d ", data->touch_protocol_type);
+
+		ret = i2c_reg_read_byte_dt(&config->bus, 0xf0, &buffer);
+		if (ret < 0) {
+			LOG_ERR("read sensing mode error (%d)", ret);
+			return ret;
+		}
+		data->sensing_mode = (buffer & ONE_D_SENSING_CONTROL_BMSK);
+		LOG_DBG("sensing mode = %d", data->sensing_mode);
+	}
+	return 0;
+}
+
+static int cf1133_get_max_touches(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int ret = 0;
+	uint8_t buffer;
+
+	ret = i2c_reg_read_byte_dt(&config->bus, MAX_NUM_TOUCHES, &buffer);
+	if (ret < 0) {
+		LOG_ERR("read sensing mode error (%d)", ret);
+		return ret;
+	}
+	data->max_touches = buffer;
+	if (data->max_touches > SITRONIX_MAX_SUPPORTED_POINT) {
+		data->max_touches = SITRONIX_MAX_SUPPORTED_POINT;
+	}
+	LOG_DBG("max touches = %d ", data->max_touches);
+
+	return 0;
+}
+
+static int cf1133_ts_init(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int r;
+	/* Get device status before use to do once */
+	r = cf1133_get_chip_id(data, config);
+	if (r < 0) {
+		LOG_ERR("Read chip id failed error %d :", r);
+		return r;
+	}
+
+	r = cf1133_get_protocol_type(data, config);
+	if (r < 0) {
+		LOG_ERR("Read protocol failed error %d :", r);
+		return r;
+	}
+
+	r = cf1133_get_max_touches(data, config);
+	if (r < 0) {
+		LOG_ERR("Read max touches failed error %d :", r);
+		return r;
+	}
+
+	if (data->touch_protocol_type == SITRONIX_A_TYPE) {
+		data->pixel_length = PIXEL_DATA_LENGTH_A;
+		LOG_DBG("pixel length : %d", data->pixel_length);
+	} else {
+		data->pixel_length = PIXEL_DATA_LENGTH_B;
+		data->max_touches = 2;
+		LOG_DBG("pixel length :  %d", data->pixel_length);
+		LOG_DBG("max touches (revised) = %d", data->max_touches);
+	}
+
+	return r;
+}
+
+static int cf1133_process(const struct device *dev)
+{
+	const struct cf1133_config *config = dev->config;
+	struct cf1133_data *data = dev->data;
+	uint16_t x, y;
+	int r;
+	uint8_t buffer[1 + MAX_TOUCHES_SPE * PIXEL_DATA_LENGTH_A] = {0};
+
+	/* Coordinates are retrieved for one or more valid touch points detected*/
+	r = i2c_burst_read_dt(&config->bus, KEYS_REG, buffer,
+			      1 + data->max_touches * data->pixel_length);
+	if (r < 0) {
+		LOG_ERR("Read Keys and coordinate registers (XY0 H/L) failed");
+		return r;
+	}
+
+	/*Coordinates for only one valid touch points are being taken into account*/
+	if (buffer[1 + 0 * data->pixel_length + XY_COORD_H] & 0x80) {
+		x = (uint16_t)(buffer[1 + 0 * data->pixel_length + XY_COORD_H] & 0x70) << 4 |
+		    buffer[1 + 0 * data->pixel_length + X_COORD_L];
+		y = (uint16_t)(buffer[1 + 0 * data->pixel_length + XY_COORD_H] & 0x07) << 8 |
+		    buffer[1 + 0 * data->pixel_length + Y_COORD_L];
```

Comment:
Can you drop the various `0 * data->pixel_length`? Clearly they don't do much.

Reply:
Ok done